go-controller/hybrid-overlay/pkg/controller/ovn_node_linux.go

```diff
@@ -227,39 +231,43 @@ func (n *NodeController) AddNode(node *kapi.Node) error {
 	klog.Info("Add Node ", node.Name)
 	var err error
 
-	if n.drIP == nil {
-		hybridOverlayDRIP, ok := node.Annotations[hotypes.HybridOverlayDRIP]
-		if !ok {
-			return fmt.Errorf("hybrid overlay not initialized on %s, it was not assigned an interface address", node.Name)
-		}
-		drIP := net.ParseIP(hybridOverlayDRIP)
-		if drIP == nil {
-			return fmt.Errorf("hybrid overlay not initialized on %s, the the annotation %s = %s is not an IP address", node.Name, hotypes.HybridOverlayDRIP, hybridOverlayDRIP)
-		}
-	}
-
 	if node.Name == n.nodeName {
+		// Add local node and setup DR
 		// Retry hybrid overlay initialization if the master was
 		// slow to add the hybrid overlay logical network elements
 		err = n.EnsureHybridOverlayBridge(node)
-	} else {
-		klog.Infof("Add hybridOverlay Node %s", node.Name)
-		err = n.hybridOverlayNodeUpdate(node)
-	}
-	if atomic.LoadUint32(n.initState) == hotypes.DistributedRouterInitialized {
-		pods, err := n.localPodLister.List(labels.Everything())
 		if err != nil {
-			return fmt.Errorf("cannot fully initialize node %s for hybrid overlay, cannot list pods: %v", n.nodeName, err)
+			return err
 		}
-		for _, pod := range pods {
-			err := n.AddPod(pod)
+	}
+
+	if atomic.LoadUint32(n.initState) >= hotypes.DistributedRouterInitialized {
+		if node.Name != n.nodeName {
+			// add remote node
+			klog.Infof("Add hybridOverlay remote Node %s", node.Name)
+			err = n.hybridOverlayNodeUpdate(node)
+			if err != nil {
+				return err
+			}
+		} else if atomic.LoadUint32(n.initState) < hotypes.PodsInitialized {
+			// add pods local to our node
+			pods, err := n.localPodLister.List(labels.Everything())
 			if err != nil {
-				klog.Errorf("Cannot wire pod %s for hybrid overlay, %v", pod.Name, err)
+				return fmt.Errorf("cannot fully initialize node %s for hybrid overlay, cannot list pods: %v", n.nodeName, err)
 			}
+			for _, pod := range pods {
+				if pod.Spec.NodeName != n.nodeName {
```

Comment:
@JacobTanenbaum asked me to add a comment here to indicate that since we went to IC the local pod informer no longer only gets local pods.